### ISSUE TITLE
[sparksql/databricks] False alert of syntax error: IS [NOT] DISTINCT FROM is unparsable

### ISSUE THREAD

**Issue by Jingyi-ZHOU** [sqlfluff/sqlfluff#4733]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have a sql as 
```
SELECT
    TS.TEST_TABLE.TEST_IS AS ID,
    CAST(TS.TEST_TABLE.CREATED_AT AS DATE) AS DY
FROM
    TS.TEST_TABLE
WHERE TS.TEST_TABLE.BOOL_COL IS NOT DISTINCT FROM true
GROUP BY 1, 2
```
which is supported by standard Spark SQL and Databricks.
But after I ran `sqlfluff lint test.sql --dialect databricks`, I got 
```
L:   6 | P:  29 |  PRS | Line 6, Position 29: Found unparsable section: ' IS NOT
                       | DISTINCT FROM true
```

### Expected Behaviour

expected behavior is syntax check passed.

### Observed Behaviour

Observed behavior is syntax check failed.

### How to reproduce

sql is
```
SELECT
    TS.TEST_TABLE.TEST_IS AS ID,
    CAST(TS.TEST_TABLE.CREATED_AT AS DATE) AS DY
FROM
    TS.TEST_TABLE
WHERE TS.TEST_TABLE.BOOL_COL IS NOT DISTINCT FROM true
GROUP BY 1, 2
```
I saved it as test.sql and then ran 
`sqlfluff lint test.sql --dialect databricks`
and
`sqlfluff lint test.sql --dialect sparksql`

### Dialect

sparksql and databricks

### Version

sqlfluff, version 2.0.3

### Configuration

all default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by friendofasquid** [comment]
Similar issue with Snowflake dialect. This is reported as unparseable:

```sql
select
    case
        when false is distinct from null then 1
    end
```
